PR: https://github.com/odoo/odoo/pull/96551

From: 0a8c3e745127374a8bf53f3b5931384c0a96978f
From: Dossogne Bertrand
Date: 2022-09-07 21:10:25

Structural Changes: 1
Total Changes: 38

[IMP] hr_*: improve web application submission

hr_*:
hr
hr_contract
hr_recruitment
website_hr_recruitment

Increase UX and possibilities to postulate to a job on recruitement.
Also moves the model hr.contract.type from hr_contract to hr so that it
can also be used from hr_recruitment without having two models in parallel.

taskID 2898063

closes odoo/odoo#96551

Related: odoo/enterprise#29777
Related: odoo/upgrade#3861
Signed-off-by: Kevin Baptiste <kba@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_contract/models/__init__.py
+++ b/addons/hr_contract/models/__init__.py
@@ -7,4 +7,4 @@ from . import hr_contract
 from . import res_users
 from . import resource
 from . import resource_resource
-from . import hr_contract_type
+from . import hr_payroll_structure_type

--- a/addons/hr_contract/models/hr_contract_type.py
+++ b/addons/hr_contract/models/hr_payroll_structure_type.py
@@ -4,12 +4,6 @@
 from odoo import fields, models
 
 
-class ContractType(models.Model):
-    _name = 'hr.contract.type'
-    _description = 'Contract Type'
-
-    name = fields.Char(required=True, translate=True)
-
 class HrPayrollStructureType(models.Model):
     _name = 'hr.payroll.structure.type'
     _description = 'Salary Structure Type'
